### DISCUSSION TITLE
UI navigation and usability improvement ideas

### DISCUSSION THREAD

**Discussion by raunz** [ngoduykhanh/PowerDNS-Admin#1418] [Ideas]
I've been trying for several weeks to domesticate the PDA to match our needs and I'll share some ideas, where PDA is a little "short":

1. When editing a record, there will appear a button "**Save**", which is actually half-truth. To actually save the change, You have to click "Save Changes" button on the top of page. Perhaps we should be rename the "**Save**" button to "**Stage changes**" or "**Accept**" or something less misleading?
User, who is first time in PDA, won't figure it out, that You have to save single change twice.

2. **Direct link** from Global Search "Record" result to the actual RR editing modal in zone. If You have really large zones and hundreds of zones, then current navigation directs user to plain zone record listing. To find the desired RR, You have to use "Search" in the table header again. As the zones records listing table is generated by DataTables, I don't know how to "shortcut" to correct place or even correct page (if listing is paginated).

3. **A <--> PTR** and vice versa shortcuts. In the A/AAAA record editing modal there would be great help to have a shortcut/button near IP address to navigate directly to corresponding PTR record (or to zone if PTR doesn't exist but in-addr.arpa zone is present). Also the modal should show the rdata value of the opposing record.
For example while editing firstname.example.com A record pointing to 100.127.255.3 the UI would provide a link to 3.255.127.100.in-addr.arpa. RR and display out that 3.255.127.100.in-addr.arpa PTR is currently pointing to "secondname.example.com".

4. In Global Search by IP address would be great to somehow look dynamically also for arpa reverse PTRs. If the input is valid IP4/6 address create ...arpa and perform additional lookup for PTRs or arpa zone. Todays IPv6 PTR editing supports pretty_ipv6_ptr but global search doesn't.

5. ~~**Changelog auto expand** or other compact view.
Current changelog is displaying without clicking only the "Changed on" and "Changed by" values for RR or zone. It is quite annoying to click each row, to see the actual change content. Thankfully there's also a search option in the "Activity" results page. In zone Changelog with thousands of changes, this click-and-look style isn't usable. I would like to see at least the name and content of each change without clicking.~~

Otherwise the PDA will almost do :) It is still the best publicly available frontend for PowerDNS.

**Comment by WhatshallIbreaktoday** [comment]
Hello Team, 

I have some improvement idea regarding the domain table :   (Click on Dashboard, select a existing domain).
I'm speaking of the "Name" column (related to the records Names)

I think the space defined for the record names column is too small  (should be twice the actual size).

Depending on the effort, we could,

1. double it's static size
2. Add the size in parameters..?
3. Make the whole table dynamically resisable ?  (but this would need to be generalized to all templates...) maybe overkill...

What do you think ?
Best regards,
